File: packages/python/latency-engine/src/shared/types/latency_types.py
from __future__ import annotations

from dataclasses import dataclass
from datetime import date


@dataclass(frozen=True)
class PercentilesResponseDTO:
    p50: float
    p95: float
    p99: float
    sample_count: int


@dataclass(frozen=True)
class SLOResponseDTO:
    burn_fast: float
    burn_medium: float
    burn_slow: float
    budget_remaining_pct: float
    slo_threshold_ms: float


@dataclass(frozen=True)
class BaselinePointDTO:
    date: date
    p99_ttft_ms: float
    p99_total_ms: float


@dataclass(frozen=True)
class AttributionResponseDTO:
    dns: float
    tcp: float
    queue: float
    inference: float

File: packages/node/web-app/src/features/latency/types/latency.types.ts
export interface PercentilesResult {
  p50: number;
  p95: number;
  p99: number;
  sample_count: number;
}

export interface SLOResult {
  burn_fast: number;
  burn_medium: number;
  burn_slow: number;
  budget_remaining_pct: number;
  slo_threshold_ms: number;
}

export interface BaselinePoint {
  date: string;
  p99_ttft_ms: number;
  p99_total_ms: number;
}

export interface AttributionResult {
  dns: number;
  tcp: number;
  queue: number;
  inference: number;
}
